Sources/NIOExtrasPerformanceTester/main.swift

```diff
@@ -0,0 +1,139 @@
+                .wait()
+            clientChannels.append(clientChannel)
+        }
+
+        var writeFutures: [EventLoopFuture<Void>] = []
+        for clientChannel in clientChannels {
+            clientChannel.write(NIOAny(HTTPClientRequestPart.head(head)), promise: nil)
+            writeFutures.append(clientChannel.writeAndFlush(NIOAny(HTTPClientRequestPart.end(nil))))
+        }
+        let allWrites = EventLoopFuture<Void>.andAllComplete(writeFutures, on: writeFutures.first!.eventLoop)
+        try! allWrites.wait()
+
+        let streamCompletedFutures = requestHandlers.map { rh in rh.completedFuture }
+        let requestsServed = EventLoopFuture<Int>.reduce(0, streamCompletedFutures, on: streamCompletedFutures.first!.eventLoop, +)
+        reqs.append(try! requestsServed.wait())
+    }
+    return reqs.reduce(0, +) / numberOfRepeats
+}
+
+// MARK:  Tests
+measureAndPrint(desc: "http1_threaded_50reqs_500conns") {
+    testHttp1Performance(numberOfRepeats: 50,
+                         numberOfClients: System.coreCount,
+                         requestsPerClient: 500,
+                         extraInitialiser: { channel in return channel.eventLoop.makeSucceededFuture(()) })
+}
+
+measureAndPrint(desc: "http1_threaded_50reqs_500conns_rolling_pcap") {
+    func addRollingPCap(channel: Channel) -> EventLoopFuture<Void> {
+        let pcapRingBuffer = NIOPCAPRingBuffer(maximumFragments: 25,
+                                               maximumBytes: 1_000_000)
+        let pcapHandler = NIOWritePCAPHandler(mode: .client,
+                                              fileSink: pcapRingBuffer.addFragment)
+        return channel.pipeline.addHandler(pcapHandler, position: .first)
+    }
+
+    return testHttp1Performance(numberOfRepeats: 50,
+                                numberOfClients: System.coreCount,
+                                requestsPerClient: 500,
+                                extraInitialiser: { channel in return addRollingPCap(channel: channel) })
+}
+
+try! measureAndPrint(desc: "http1_threaded_50reqs_500conns_rolling_pcap") {
+    let outputFile = NSTemporaryDirectory() + "/" + ProcessInfo().globallyUniqueString
+    let fileSink = try NIOWritePCAPHandler.SynchronizedFileSink.fileSinkWritingToFile(path: outputFile) { error in
+        print("ERROR: \(error)")
+        exit(1)
+    }
+    defer {
+        try! fileSink.syncClose()
+    }
+
+    func addPCap(channel: Channel) -> EventLoopFuture<Void> {
+        let pcapHandler = NIOWritePCAPHandler(mode: .client,
+                                              fileSink: fileSink.write)
+        return channel.pipeline.addHandler(pcapHandler, position: .first)
+    }
+
+    return testHttp1Performance(numberOfRepeats: 50,
+                                numberOfClients: System.coreCount,
+                                requestsPerClient: 500,
+                                extraInitialiser: { channel in return addPCap(channel: channel) })
+}
```

Comment:
I'd prefer to see these tests use the new `Benchmark` class-based model rather than this pattern. One test per file is a lot nicer.